client/driver/rkt.go

```diff
@@ -0,0 +1,201 @@
+// planned in the future
+type RktDriver struct {
+        DriverContext
+}
+
+// rktHandle is returned from Start/Open as a handle to the PID
+type rktHandle struct {
+        proc   *os.Process
+        name   string
+        waitCh chan error
+        doneCh chan struct{}
+}
+
+// rktPID is a struct to map the pid running the process to the vm image on
+// disk
+type rktPID struct {
+        Pid  int
+        Name string
+}
+
+// NewRktDriver is used to create a new exec driver
+func NewRktDriver(ctx *DriverContext) Driver {
+        return &RktDriver{*ctx}
+}
+
+func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
+        // Only enable if we are root when running on non-windows systems.
+        if runtime.GOOS != "windows" && syscall.Geteuid() != 0 {
+                d.logger.Printf("[DEBUG] driver.rkt: must run as root user, disabling")
+                return false, nil
+        }
+
+        outBytes, err := exec.Command("rkt", "version").Output()
+        if err != nil {
+                return false, nil
+        }
+        out := strings.TrimSpace(string(outBytes))
+
+        rktMatches := reRktVersion.FindStringSubmatch(out)
+        appcMatches := reRktVersion.FindStringSubmatch(out)
+        if len(rktMatches) != 2 || len(appcMatches) != 2 {
+                return false, fmt.Errorf("Unable to parse Rkt version string: %#v", rktMatches)
+        }
+
+        node.Attributes["driver.rkt"] = "1"
+        node.Attributes["driver.rkt.version"] = rktMatches[0]
+        node.Attributes["driver.rkt.appc.version"] = appcMatches[1]
+
+        return true, nil
+}
+
+// Run an existing Rkt image.
+func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
+        trust_prefix, ok := task.Config["trust_prefix"]
+        if !ok || trust_prefix == "" {
+                return nil, fmt.Errorf("Missing trust prefix for rkt")
+        }
+        trust_args := []string{
+                "rkt",
+                "trust",
+                "--prefix=" + trust_prefix,
+        }
+        // Add the given trust prefix
+        var outBuf, errBuf bytes.Buffer
+        cmd := exec.Command(trust_args[0], trust_args[1:]...)
+        cmd.Stdout = &outBuf
+        cmd.Stderr = &errBuf
+        d.logger.Printf("[DEBUG] Starting rkt command: %q", strings.Join(trust_args, " "))
+        if err := cmd.Start(); err != nil {
+                return nil, fmt.Errorf(
+                        "Error running rkt: %s\n\nOutput: %s\n\nError: %s",
+                        err, outBuf.String(), errBuf.String())
+        }
+        d.logger.Printf("[DEBUG] Added trust prefix: %q", trust_prefix)
+
+        name, ok := task.Config["name"]
+        if !ok || name == "" {
+                return nil, fmt.Errorf("Missing ACI name for rkt")
+        }
+        run_args := []string{
+                "rkt",
+                "run", "--interactive", name,
```

Comment:
I don't think we want to use --interactive. That attaches the pod to `tty`. I think rather we want to use the `status --wait uuid` command: https://coreos.com/rkt/docs/latest/subcommands/status.html